tests/Gedmo/SoftDeleteable/CarbonTest.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/*
+ * This file is part of the Doctrine Behavioral Extensions package.
+ * (c) Gediminas Morkevicius <gediminas.morkevicius@gmail.com> http://www.gediminasm.org
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Gedmo\Tests\SoftDeleteable;
+
+use Carbon\Carbon;
+use Carbon\Doctrine\DateTimeType;
+use Doctrine\Common\EventManager;
+use Doctrine\DBAL\Types\Type as DoctrineType;
+use Doctrine\DBAL\Types\Types;
+use Gedmo\SoftDeleteable\Filter\SoftDeleteableFilter;
+use Gedmo\SoftDeleteable\SoftDeleteableListener;
+use Gedmo\Tests\SoftDeleteable\Fixture\Entity\Article;
+use Gedmo\Tests\SoftDeleteable\Fixture\Entity\Comment;
+use Gedmo\Tests\Tool\BaseTestCaseORM;
+
+final class CarbonTest extends BaseTestCaseORM
```

Comment:
IMO, these classes should not include "Carbon" in its names, but something like "CustomDateTime".

Reply:
Well, I can rename these files.

However they not only test custom field types but also check integration with Carbon converter.

`CustomDateTime` might be confusing too as `DateTime` is one of Doctrine types for physical time. Others are `Time` and `Date` (and all are postfixed by `_immutable` or `_mutable`). And custom converter should not be always of any time type. We can define a new custom type `my_timestamp` that will be stored in the database as a string or blob. And it will be handled by the same code.

So `CustomFieldType` would make more sense, but there is only one simple test there. Is it reasonable to discuss its name?

Reply:
I think these are good concerns. The goal of my suggestion is to keep these files agnostic as possible in order to allow more tests in the future for custom implementations other than Carbon.
IMO, if the integration with the Carbon converter can be made separately, I think it could be a good decision, as
it is always better to keep the responsibilities separated.

Reply:
Uhm, I don't get it. What should I do?

Reply:
>However they not only test custom field types but also check integration with Carbon converter.

>Uhm, I don't get it. What should I do?

From my perspective, the test should be organized in this way:
1. Any test related to the custom types should cover the responsibilities related to a custom type implementation (Carbon or others).
2. If we have a exclusive test covering the integration with the Carbon converter, I think this test should be provided in a separate class, as it is only related to the Carbon package.

Please, let me know if you need more context about my suggestion.

Reply:
There are only two tests (one for each extension). They are just copies of tests which check the main behavior with added field type assertions. 

In order to test this feature we need a Doctrine Field Converter. We may write a fake one by our own but we will have to maintain it and track when Doctrine change any requirements. Or we may use quite popular and maintained converter, the one provided by the Carbon team.

I've chosen the latter option.